torch/csrc/distributed/c10d/UCCUtils.hpp

```diff
@@ -31,6 +31,30 @@ namespace c10d {
     }                                     \
   } while (0)
 
+// Macro and throw on a non-successful UCC return value and free its request.
+#define TORCH_UCC_CHECK_REQUEST(_request, _cmd, _error_msg) \
+  do {                                                      \
+    ucc_status_t result = _cmd;                             \
+    if (result != UCC_OK) {                                 \
+      std::string err = c10::str(                           \
+          "[",                                              \
+          std::string(__FILE__),                            \
+          ":",                                              \
+          std::to_string(__LINE__),                         \
+          "] ",                                             \
+          logger->getLogPrefix(),                           \
+          _error_msg,                                       \
+          ", error code ",                                  \
+          result,                                           \
+          ": ",                                             \
+          ucc_status_string(result),                        \
+          ", system error code ",                           \
+          errno);                                           \
+      ucc_collective_finalize(_request);                    \
+      TORCH_CHECK(false, err);                              \
```

Comment:
I do not think we need additional check for `ucc_collective_finalize` here.

Reply:
Would we lose the original error messages, i.e. `err` and `_error_msg`, if we `TORCH_UCC_CHECK`'ing `ucc_collective_finalize` here?

Reply:
You are right, if `ucc_collective_finalize` is failed then the iriginal message `err` will be lost.